pkg/replicaset/replicaset.go

```diff
@@ -0,0 +1,469 @@
+package replicaset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Builder provides struct for replicaset object containing connection to the cluster and the replicaset definitions.
+type Builder struct {
+	// Replicaset definition. Used to create a replicaset object.
+	Definition *appsv1.ReplicaSet
+	// Created replicaset object.
+	Object *appsv1.ReplicaSet
+	// Used in functions that define or mutate replicaset definition. errorMsg is processed before the replicaset
+	// object is created.
+	errorMsg  string
+	apiClient *clients.Settings
+}
+
+// AdditionalOptions additional options for replicaset object.
+type AdditionalOptions func(builder *Builder) (*Builder, error)
+
+var retryInterval = time.Second * 3
+
+// NewBuilder creates a new instance of Builder.
+func NewBuilder(
+	apiClient *clients.Settings,
+	name, nsname string,
+	labels map[string]string,
+	replicas int32,
```

Comment:
Please remove replicas. Replica will be set automatically.
The logic of NewBuilder is to provide only **minimum** required parameters for an object. 
Please refer to: https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/#replicas